core/App/screens/Connection.tsx

```diff
@@ -54,71 +62,78 @@ const Connection: React.FC<ConnectionProps> = ({ navigation, route }) => {
     },
   })
 
+  const setModalVisible = (value: boolean) => {
+    setState((prev) => ({ ...prev, isVisible: value }))
+  }
+  const setIsInitialized = (value: boolean) => {
+    setState((prev) => ({ ...prev, isInitialized: value }))
+  }
+  const setShouldShowDelayMessage = (value: boolean) => {
+    setState((prev) => ({ ...prev, shouldShowDelayMessage: value }))
+  }
+  const abortTimer = () => {
+    if (timerRef.current) {
+      clearTimeout(timerRef.current)
+      timerRef.current = null
+    }
+  }
   const onDismissModalTouched = () => {
-    timer && clearTimeout(timer)
-
-    navigation.getParent()?.navigate(TabStacks.HomeStack, { screen: Screens.Home })
-
+    abortTimer()
```

Comment:
Needed? I though the return value from useFocusEffect would cleanup - no?

Reply:
You are right, `useFocusEffect` will clean up. I've removed the call from here.